components/user-store/org.wso2.carbon.identity.user.store.configuration/src/main/java/org/wso2/carbon/identity/user/store/configuration/utils/DefaultUserStoreAttributeMappingParser.java

```diff
@@ -0,0 +1,160 @@
+
+        StAXOMBuilder builder;
+        OMElement localClaimElement = null;
+        try {
+            builder = new StAXOMBuilder(inStream);
+            Iterator iterator = builder.getDocumentElement().
+                    getFirstChildWithName(new QName(DIALECTS)).
+                    getChildrenWithLocalName(DIALECT);
+            if (iterator != null) {
+                // Select local claim attributes.
+                while (iterator.hasNext()) {
+                    localClaimElement = (OMElement) iterator.next();
+                    Iterator attributeIterator = localClaimElement.getAllAttributes();
+                    if (attributeIterator != null) {
+                        String attributeValue = ((OMAttribute) attributeIterator.next()).getAttributeValue();
+                        if (StringUtils.equalsIgnoreCase(LOCAL_DIALECT_URL, attributeValue)) {
+                            break;
+                        }
+                    }
+                }
+                if (localClaimElement == null) {
+                    if (LOG.isDebugEnabled()) {
+                        LOG.debug("Local claims cannot be found.");
+                    }
+                    return;
+                }
+                Iterator claimIterator = localClaimElement.getChildElements();
+                defaultAttributes = new HashMap<>();
+                while (claimIterator.hasNext()) {
+                    OMElement claimElement = (OMElement) claimIterator.next();
+                    Iterator attributeIterator = claimElement.getChildElements();
+                    UserStoreAttributeDO userStoreAttributeDO = new UserStoreAttributeDO();
+                    while (attributeIterator.hasNext()) {
+                        OMElement attributes = (OMElement) attributeIterator.next();
+                        String attributeQName = attributes.getQName().getLocalPart();
+                        if (StringUtils.equalsIgnoreCase(DISPLAY_NAME, attributeQName)) {
+                            userStoreAttributeDO.setDisplayName(attributes.getText());
+                        } else if (StringUtils.equalsIgnoreCase(ATTRIBUTE_ID, attributeQName)) {
+                            userStoreAttributeDO.setMappedAttribute(attributes.getText());
+                        } else if (StringUtils.equalsIgnoreCase(CLAIM_URI, attributeQName)) {
+                            userStoreAttributeDO.setClaimUri(attributes.getText());
+                            userStoreAttributeDO.setClaimId(Base64.getUrlEncoder().withoutPadding().
+                                    encodeToString(attributes.getText().getBytes(StandardCharsets.UTF_8)));
+                        }
+                    }
+                    defaultAttributes.put(userStoreAttributeDO.getClaimId(), userStoreAttributeDO);
+                }
+            }
+        } catch (XMLStreamException e) {
+            LOG.error("Error occurred while reading the claim-config.xml file.", e);
```

Comment:
Is it safe to continue from here after this error ? Don't we need to communicate this error to upper layers ?